errcheck/creator/creator.go

```diff
@@ -39,11 +44,24 @@ func Errcheck() checker.Creator {
 			}
 
 			var args []string
+			var cleanup func()
+
 			if len(cfg.Ignore) > 0 {
-				args = append(args, "-ignore")
-				args = append(args, cfg.Ignore...)
+				args = append(args, "-ignore", strings.Join(cfg.Ignore, ","))
+			}
+			if len(cfg.Exclude) > 0 {
+				excludeFile, err := createExcludeFile(cfg.Exclude)
```

Comment:
As it stands right now, if the program is killed after this checker is created but before it is run, it won't run the cleanup operation and this file will be persisted. If you use the `ContextWithShutdown` approach described below and hook that up here, you can ensure that the cleanup is run after execution OR if app is shut down before regular cleanup. It may require some minor restructuring of the cleanup code, but think it should be reasonable to do and is more correct.